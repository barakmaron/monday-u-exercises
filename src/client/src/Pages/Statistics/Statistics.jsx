import { React, useState } from 'react';
import { useStatistics } from '../../Hooks/useStatustics';
import { Chart as ChartJS, 
  ArcElement, 
  Tooltip, 
  Legend, 
  Title, 
  CategoryScale,
  LinearScale,
  BarElement } from 'chart.js';
import { Pie, Bar } from 'react-chartjs-2';
import { Heading, Divider } from 'monday-ui-react-core';
import StatisticsInitObjects from '../../globals/StatisticsInitObjects';
import style from './statistics.module.css';

ChartJS.register(ArcElement, 
  Tooltip, 
  Legend);
ChartJS.register(
  CategoryScale,
  LinearScale,
  BarElement,
  Title,
  Tooltip,
  Legend
);

function Statistics() {
  const [pie_created_data, setPieCreatedData] = useState(StatisticsInitObjects.CREATED_OBJECT);
  const [bar_pokemon_data, setBarPokemonData] = useState(StatisticsInitObjects.POKEMON_OBJECT);

  const [completed, setCompleted] = useState(StatisticsInitObjects.COMPLETED_OBJECT);
  useStatistics(setPieCreatedData, setBarPokemonData, setCompleted);
  return ( <div className={style.statistics_main_container}>
      <div className={style.statistics_container}>
        <Heading type={Heading.types.h2} value="Number of todos created in each day" id="created_pie_h2"/>
        <div >
          <Pie data={pie_created_data}/>
        </div>
      </div>
      <div className={style.statistics_container}>
        <Heading type={Heading.types.h2} value="Number of pokemon vs regular tasks" id="pokemon_bar_h2"/>
        <div>
          <Bar data={bar_pokemon_data}/>
        </div>
      </div>
      <Divider direction={Divider.directions.HORIZONTAL} />
      <div className={style.statistics_container}>
        <Heading type={Heading.types.h2} value="Completed time statistics" id="completed_h2"/>      
        <div>
          <Heading type={Heading.types.h3} value="Max time:" id="max_time_h3"/>
          {completed.max.days} days {completed.max.hours} hours {completed.max.minutes} minuets
        </div>
        <div>
          <Heading type={Heading.types.h3} value="Min time:" id="Min_time_h3"/>
          {completed.min.days} days {completed.min.hours} hours {completed.min.minutes} minuets
        </div>
        <div>
          <Heading type={Heading.types.h3} value="Average time:" id="max_time_h3"/>
          {completed.average.days} days {completed.average.hours} hours {completed.average.minutes} minuets
        </div>
        <div>
          <Heading type={Heading.types.h3} value="Number of completed todos:" id="number_completed_h3"/>
          {completed.number_completed} todos
        </div>
      </div>
  </div>);
};

export default Statistics;
