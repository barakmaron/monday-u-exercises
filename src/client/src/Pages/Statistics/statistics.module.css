.statistics_main_container {
    display: flex;
    width: -webkit-fill-available; 
    flex-wrap: wrap;
    flex-direction: row;
    justify-content: center;
    align-items: flex-start;
    align-content: space-between;
}

.statistics_container{
    margin: 10PX 10px;
    width: 45%;
    display: flex;
    flex-direction: column;
    align-items: center;
    align-content: center;
    text-align: center;
    
}