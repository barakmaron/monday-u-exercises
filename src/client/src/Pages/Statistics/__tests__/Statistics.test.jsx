import renderer from 'react-test-renderer';
import Statistics from '../Statistics';

it('Render snapshot of StatisticsPage', () => {
    const tree = renderer.create(<Statistics></Statistics>).toJSON();
    expect(tree).toMatchSnapshot();
});