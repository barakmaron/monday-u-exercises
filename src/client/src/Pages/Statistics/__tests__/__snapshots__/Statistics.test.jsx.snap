// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render snapshot of StatisticsPage 1`] = `
<div
  className="statistics_main_container"
>
  <div
    className="statistics_container"
  >
    <h2
      aria-label=""
      className="heading-component element-type-h2 size-large single-line-ellipsis"
      id="created_pie_h2"
    >
      Number of todos created in each day
    </h2>
    <div>
      <canvas
        height={150}
        role="img"
        width={300}
      />
    </div>
  </div>
  <div
    className="statistics_container"
  >
    <h2
      aria-label=""
      className="heading-component element-type-h2 size-large single-line-ellipsis"
      id="pokemon_bar_h2"
    >
      Number of pokemon vs regular tasks
    </h2>
    <div>
      <canvas
        height={150}
        role="img"
        width={300}
      />
    </div>
  </div>
  <div
    className="monday-style-divider monday-style-divider--horizontal"
  />
  <div
    className="statistics_container"
  >
    <h2
      aria-label=""
      className="heading-component element-type-h2 size-large single-line-ellipsis"
      id="completed_h2"
    >
      Completed time statistics
    </h2>
    <div>
      <h3
        aria-label=""
        className="heading-component element-type-h3 size-large single-line-ellipsis"
        id="max_time_h3"
      >
        Max time:
      </h3>
      0
       days 
      0
       hours 
      0
       minuets
    </div>
    <div>
      <h3
        aria-label=""
        className="heading-component element-type-h3 size-large single-line-ellipsis"
        id="Min_time_h3"
      >
        Min time:
      </h3>
      0
       days 
      0
       hours 
      0
       minuets
    </div>
    <div>
      <h3
        aria-label=""
        className="heading-component element-type-h3 size-large single-line-ellipsis"
        id="max_time_h3"
      >
        Average time:
      </h3>
      0
       days 
      0
       hours 
      0
       minuets
    </div>
    <div>
      <h3
        aria-label=""
        className="heading-component element-type-h3 size-large single-line-ellipsis"
        id="number_completed_h3"
      >
        Number of completed todos:
      </h3>
      0
       todos
    </div>
  </div>
</div>
`;
